app/jobs/generate_pdf_job.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'open3'
+require 'pathname'
+
+# Background job to generate a PDF from EAD metadata
+class GeneratePdfJob < ApplicationJob
+  class GeneratePdfError < StandardError; end
+
+  # There are very few situations where a PDF generation job would fail but then
+  # succeed in a subsequent retry.
+  sidekiq_options retry: 1
+
+  # Generate PDFs for all XML files contained within a directory (deeply).
+  # PDF files will be created in the same directory as the XML and will share the same name.
+  # For example `ead123.xml` would result in `ead123.pdf`.
+  # @param data_dir [String] the path to the directory containing the XML files
+  # @param skip_existing [Boolean] if true, XML files that have existing corresponding PDFs will be skipped
+  # rubocop:disable Metrics/AbcSize
+  def self.enqueue_all(data_dir: Settings.data_dir, skip_existing: false)
+    fop_path = Settings.pdf_generation.fop_path || 'fop'
+
+    Dir.glob("#{data_dir}/**/*.xml") do |file_path|
+      file_dir = File.dirname(file_path)
+      file_name = File.basename(file_path)
+      GeneratePdfJob.perform_later(file_path:, file_name:, data_dir: file_dir, skip_existing:, fop_path:,
+                                   saxon_path: Settings.pdf_generation.saxon_path,
+                                   fop_config_path: Rails.root.join('config/pdf_generation/fop-config.xml').to_s,
+                                   ead_to_fo_xsl_path: Rails.root.join('app/xslt/ead_to_fo.xsl').to_s,
+                                   logo_path: Rails.root.join('app/assets/images/sul-logo-bw_lightbackground.png').to_s)
+    end
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  # Generates a PDF for a given EAD XML file. Will overwrite any existing PDFs with the same name.
+  # @param file_path [String] the path of the XML file, such as '/opt/app/arclight/data/ars/ars0001.xml'
+  # @param file_name [String] the name of the PDF file to be created, such as 'ead1234'
+  # @param data_dir [String] the path where the file should be saved, such as '/opt/app/arclight/data/ars'
+  # @param skip_existing [Boolean] if true, an existing PDF will not be replaced
+  # @param fop_path [String] the path to the Apache FOP executable
+  # @param fop_config_path [String] the path to the Apache FOP configuration XML file
+  # @param ead_to_fo_xsl_path [String] the path to the XSLT file to use to convert EAD to FO
+  # @param saxon_path [String] the path to the Saxon jar file
+  # @param logo_path [String] the path to the logo to use in the PDF
+  # rubocop:disable Metrics/ParameterLists
+  def perform(file_path:, file_name:, data_dir:, skip_existing:, fop_path:,
```

Comment:
Is there a reason to prefer passing the various paths to installed resources vs. using what's set in the config directly within the method?

Reply:
@corylown actually this was a bit of a puzzle to me. I had it written this way originally which worked great during testing but when running the job using sidekiq (locally), the workers were complaining that `Settings` wasn't available, so I switched to this laundry list. Should Settings be available in this case? I can take another pass at it.

Reply:
@corylown I must have been doing something wrong. After double checking that I was reloading rails/sidekiq and clearing out the old jobs, I am able to access the Settings in the Sidekiq jobs. I've altered the branch to remove all the unnecessary path passing.